Lib/pathlib.py

```diff
@@ -1071,15 +1069,18 @@ def _make_child_relpath(self, part):
         return self._from_parsed_parts(self._drv, self._root, parts)
 
     def __enter__(self):
-        if self._closed:
-            self._raise_closed()
         return self
 
     def __exit__(self, t, v, tb):
-        self._closed = True
-
-    def _raise_closed(self):
-        raise ValueError("I/O operation on closed path")
+        # https://bugs.python.org/issue39682
+        # In previous versions of pathlib, this method marked this path as
+        # closed; subsequent attempts to perform I/O would raise an IOError.
+        # This functionality was never documented, and had the effect of
+        # making Path objects mutable, contrary to PEP 428. In Python 3.9 the
+        # _closed attribute was removed, and this method made a no-op.
+        # This method and __enter__()/__exit__() should be deprecated and
+        # removed in the future.
+        pass
```

Comment:
Should this raise a `DeprecationWarning`?

Reply:
I think so, but I vaguely remember people on the issue arguing against it. I don't remember where the conversation landed.

Reply:
@pitrou any thoughts?

Reply:
Sorry for the delay. I think it's pointless to raise a `DeprecationWarning`. It's a trivial zero-cost "feature", we may as well keep it eternally (or, at least, until Python's or `pathlib`'s own demise).